packages/context/src/context.ts

```diff
@@ -47,22 +47,32 @@ export class Context {
   bind<ValueType = BoundValue>(
     key: BindingAddress<ValueType>,
   ): Binding<ValueType> {
+    const binding = new Binding<ValueType>(key.toString());
+    this.add(binding);
+    return binding;
+  }
+
+  /**
+   * Add a binding to the context. If a locked binding already exists with the
+   * same key, an error will be thrown.
+   * @param binding The configured binding to be added
+   */
+  add<ValueType = BoundValue>(binding: Binding<ValueType>): this {
```

Comment:
A nitpick: I think `= BoundValue` is not needed, because the specialization of `add` can be always inferred by the compiler from the specialization of the `binding` argument.

Reply:
@raymondfeng PTAL - is the default value for `ValueType` needed?

Reply:
I simplified it as `add(binding: Binding<unknown>): this {`. The same is also applied to a few other methods.